### PR TITLE
Lasb 1837 crime application adaptor service

### PR DESCRIPTION
**Pull request by skasthuri36326**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/LASB-1837)

Describe what you did and why.

## Checklist

Before you ask people to review this PR:

- [ ] Tests should be passing: `./gradlew test`
- [ ] Github should not be reporting conflicts; you should have recently run `git rebase main`.
- [ ] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [ ] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [ ] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: crime-applications-adaptor/src/main/java/uk/gov/justice/laa/crime/applications/adaptor/util/CrimeApplicationUtil.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import io.jsonwebtoken.io.Decoders;
+import io.jsonwebtoken.security.Keys;
+import lombok.experimental.UtilityClass;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+@UtilityClass
+public class CrimeApplicationUtil {
```
Comment (stone-dj) [inline crime-applications-adaptor/src/main/java/uk/gov/justice/laa/crime/applications/adaptor/util/CrimeApplicationUtil.java:14]: Maybe rename this class to `CrimeApplicationHttpUtil` to make it's responsibility clearer and more specific.

File: crime-applications-adaptor/src/test/java/uk/gov/justice/laa/crime/applications/adaptor/config/MockServicesConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package uk.gov.justice.laa.crime.applications.adaptor.config;
+
+public class MockServicesConfiguration {
+
+    public static final String CLIENT_SECRET = "TESTVjBtRXlpQGNDJGZIKU5MQUVtd2NPY0FbLVN6JGg=";
```
Comment (stone-dj) [inline crime-applications-adaptor/src/test/java/uk/gov/justice/laa/crime/applications/adaptor/config/MockServicesConfiguration.java:5]: Can probably make these constants private

File: crime-applications-adaptor/src/test/java/uk/gov/justice/laa/crime/applications/adaptor/testutils/FileUtils.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+
+public class FileUtils {
+
+    public static final String CHARACTER_ENCODING = "UTF-8";
```
Comment (stone-dj) [inline crime-applications-adaptor/src/test/java/uk/gov/justice/laa/crime/applications/adaptor/testutils/FileUtils.java:12]: Can make this constant `private`.

**Review by stone-dj — CHANGES_REQUESTED** [review]
Changes look good, please action review feedback and let me know if you have any questions.

File: crime-applications-adaptor/src/main/java/uk/gov/justice/laa/crime/applications/adaptor/exception/GlobalExceptionHandler.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+import java.net.URI;
+import java.time.Instant;
+
+@RestControllerAdvice
+@Slf4j
+public class GlobalExceptionHandler {
+
+    public static final String CLIENT_ERROR = "Client_Error";
```
[the remark this answers is not in the document]
Comment (skasthuri36326) [reply crime-applications-adaptor/src/main/java/uk/gov/justice/laa/crime/applications/adaptor/exception/GlobalExceptionHandler.java]: Updated with private

File: crime-applications-adaptor/src/main/java/uk/gov/justice/laa/crime/applications/adaptor/util/CrimeApplicationUtil.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    public static final String AUTHORIZATION = "Authorization";
```
[the remark this answers is not in the document]
Comment (skasthuri36326) [reply crime-applications-adaptor/src/main/java/uk/gov/justice/laa/crime/applications/adaptor/util/CrimeApplicationUtil.java]: Updated with private

File: crime-applications-adaptor/src/test/java/uk/gov/justice/laa/crime/applications/adaptor/config/MockServicesConfiguration.java (hunk 2 above)
Comment (skasthuri36326) [reply crime-applications-adaptor/src/test/java/uk/gov/justice/laa/crime/applications/adaptor/config/MockServicesConfiguration.java]: Updated to private

File: crime-applications-adaptor/src/test/java/uk/gov/justice/laa/crime/applications/adaptor/testutils/FileUtils.java (hunk 3 above)
Comment (skasthuri36326) [reply crime-applications-adaptor/src/test/java/uk/gov/justice/laa/crime/applications/adaptor/testutils/FileUtils.java]: updated to private

**Review by stone-dj — APPROVED** [review]
Changes look good

**Pull request merged**